frontend/catalyst/compiler.py

```diff
@@ -355,15 +361,18 @@ def run_from_ir(
         if self.options.verbose:
             print(f"[LIB] Running compiler driver in {workspace}", file=self.options.logfile)
 
-        compiler_output = run_compiler_driver(
-            ir,
-            workspace,
-            module_name,
-            keep_intermediate=self.options.keep_intermediate,
-            verbose=self.options.verbose,
-            pipelines=pipelines,
-            lower_to_llvm=lower_to_llvm,
-        )
+        try:
+            compiler_output = run_compiler_driver(
+                ir,
+                workspace,
+                module_name,
+                keep_intermediate=self.options.keep_intermediate,
+                verbose=self.options.verbose,
+                pipelines=pipelines,
+                lower_to_llvm=lower_to_llvm,
+            )
+        except RuntimeError as e:
+            raise CompileError(*e.args) from e
```

Comment:
Is there a particular advantage to change this from a `RuntimeError` to a `CompileError` error? Just curious,

Reply:
I read the first point of the task description as a direct instruction to raise CompilerError. But I think I might be wrong, will clarify this